src/QsCompiler/DataStructures/SyntaxTree.fs

```diff
@@ -789,11 +791,91 @@ type QsNamespace = {
+    /// whether they are called, partially applied, or assigned. Each key in the returned lookup
+    /// represents a specialization that is used by the caller. Each value in the lookup is an
+    /// IEnumerable of edges representing all the different ways the given caller specialization took a
+    /// dependency on the specialization represented by the associated key.
+    /// Throws ArgumentNullException if argument is null.
+    /// Returns an empty ILookup if the node was found with no dependencies or was not found in
+    /// the graph.
+    abstract member GetAllDependencies : ICallGraphNode -> ILookup<ICallGraphNode, ICallGraphEdge>
+    /// Returns all specializations that are used directly or indirectly within the given caller,
+    /// whether they are called, partially applied, or assigned. Each key in the returned lookup
+    /// represents a specialization that is used by the caller. Each value in the lookup is an
+    /// IEnumerable of edges representing all the different ways the given caller specialization took a
+    /// dependency on the specialization represented by the associated key.
+    /// Throws ArgumentNullException if argument is null.
+    /// Returns an empty ILookup if the node was found with no dependencies or was not found in
+    /// the graph.
+    abstract member GetAllDependencies : QsSpecialization -> ILookup<ICallGraphNode, ICallGraphEdge>
+    /// Returns all specializations that are used directly within the given caller, whether they are
+    /// called, partially applied, or assigned. Each key in the returned lookup represents a
+    /// specialization that is used by the caller. Each value in the lookup is an IEnumerable of edges
+    /// representing all the different ways the given caller specialization took a dependency on the
+    /// specialization represented by the associated key.
+    /// Throws ArgumentNullException if argument is null.
+    /// Returns an empty ILookup if the node was found with no dependencies or was not found in
+    /// the graph.
+    abstract member GetDirectDependencies : ICallGraphNode -> ILookup<ICallGraphNode, ICallGraphEdge>
+    /// Returns all specializations that are used directly within the given caller, whether they are
+    /// called, partially applied, or assigned. Each key in the returned lookup represents a
+    /// specialization that is used by the caller. Each value in the lookup is an IEnumerable of edges
+    /// representing all the different ways the given caller specialization took a dependency on the
+    /// specialization represented by the associated key.
+    /// Throws ArgumentNullException if argument is null.
+    /// Returns an empty ILookup if the node was found with no dependencies or was not found in
+    /// the graph.
+    abstract member GetDirectDependencies : QsSpecialization -> ILookup<ICallGraphNode, ICallGraphEdge>
+    /// Given a call graph edges, finds all cycles and determines if each is valid.
```

Comment:
This is a minor nitpick but could you put a blank line between the members? This is a lot of text. :)

Reply:
These interfaces have been removed.